android/java/org/chromium/chrome/browser/ntp/SponsoredBackgroundWebView.java

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.ntp;
+
+import android.app.Activity;
+import android.net.Uri;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+import org.chromium.base.version_info.VersionInfo;
+import org.chromium.chrome.browser.content.WebContentsFactory;
+import org.chromium.chrome.browser.ntp_background_images.model.SponsoredRichMedia;
+import org.chromium.chrome.browser.profiles.Profile;
+import org.chromium.components.embedder_support.view.ContentView;
+import org.chromium.components.thinwebview.ThinWebView;
+import org.chromium.components.thinwebview.ThinWebViewConstraints;
+import org.chromium.components.thinwebview.ThinWebViewFactory;
+import org.chromium.content_public.browser.LoadUrlParams;
+import org.chromium.content_public.browser.WebContents;
+import org.chromium.net.NetId;
+import org.chromium.ui.base.IntentRequestTracker;
+import org.chromium.ui.base.ViewAndroidDelegate;
+import org.chromium.ui.base.WindowAndroid;
+
+public class SponsoredBackgroundWebView {
+    private WebContents mSponsoredBackgroundWebContents;
+    private ThinWebView mSponsoredBackgroundWebView;
+
+    private static final String NEW_TAB_TAKEOVER_URL = "chrome://new-tab-takeover";
+    private static final String WALLPAPER_URL = "wallpaperUrl";
+    private static final String CREATIVE_INSTANCE_ID = "creativeInstanceId";
+    private static final String PLACEMENT_ID = "placementId";
+    private static final String TARGET_URL = "targetUrl";
+
+    public SponsoredBackgroundWebView(Activity activity, WindowAndroid windowAndroid, Profile profile) {
+        mSponsoredBackgroundWebContents =
+                WebContentsFactory.createWebContentsWithWarmRenderer(
+                        profile,
+                        /* initiallyHidden= */ false,
+                        /* targetNetwork= */ NetId.INVALID);
+
+        final ContentView webContentView =
+                ContentView.createContentView(activity, mSponsoredBackgroundWebContents);
+        final ViewAndroidDelegate delegate =
+                ViewAndroidDelegate.createBasicDelegate(webContentView);
+        mSponsoredBackgroundWebContents.setDelegates(
+                VersionInfo.getProductVersion(),
+                delegate,
+                webContentView,
+                windowAndroid,
+                WebContents.createDefaultInternalsHolder());
+
+        final IntentRequestTracker intentRequestTracker = windowAndroid.getIntentRequestTracker();
```

Comment:
why `intentRequestTracker` has to be `final`, could `windowAndroid.getIntentRequestTracker()` passed directly to `create`?

Reply:
the same about `final ContentView webContentView`, `final ViewAndroidDelegate delegate`

Reply:
`intentRequestTracker` and `final ViewAndroidDelegate delegate` are fixed. 
I left `final ContentView webContentView` as it is because it is used in two places.